worldedit-bukkit/src/main/java/com/boydti/fawe/bukkit/adapter/mc1_14/BukkitGetBlocks_1_14.java

```diff
@@ -620,7 +621,38 @@ public boolean trim(boolean aggressive) {
         if (aggressive) {
             sections = null;
             nmsChunk = null;
+            return super.trim(true);
+        } else {
+            for (int i = 0; i < 16; i++) {
+                if (!hasSection(i) || super.sections[i] == CharBlocks.EMPTY) {
+                    continue;
+                }
+                ChunkSection existing = getSections()[i];
+                try {
+                    final DataPaletteBlock<IBlockData> blocksExisting = existing.getBlocks();
+
+                    final DataPalette<IBlockData> palette = (DataPalette<IBlockData>) BukkitAdapter_1_14.fieldPalette.get(blocksExisting);
+                    int paletteSize;
+
+                    if (palette instanceof DataPaletteLinear) {
+                        paletteSize = ((DataPaletteLinear<IBlockData>) palette).b();
+                    } else if (palette instanceof DataPaletteHash) {
+                        paletteSize = ((DataPaletteHash<IBlockData>) palette).b();
+                    } else {
+                        super.trim(false, i);
+                        continue;
+                    }
+                    if (paletteSize == 1) {
+                        //If the cached palette size is 1 then no blocks can have been changed i.e. do not need to update these chunks.
+                        continue;
+                    }
+                    super.trim(false, i);
+                } catch (IllegalAccessException ignored) {
```

Comment:
I would suggest, for clarity, to rename this exception field since you're not really ignoring it when you print its stacktrace.